frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.html

```diff
@@ -96,9 +96,32 @@ <h5 class="fw-light">Challenge Settings</h5>
             [phaseSelected]="phaseSelected()"
             #phaseselect
           ></app-selectphase>
-         </div>
-      </div>
+
+        <div class="row margin-bottom-cancel phase-card" *ngIf="selectedPhase">
+          <div class="col s12">
+            <button
+              class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-dark fs-12"
+              (click)="editPhaseDetails()"
+            >
+            Phase Details
+            </button>
+            <button
+              class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-dark fs-12"
+              (click)="togglePhaseVisibility()"
+            >
+              <i class="{{ phaseVisibility.icon }}"></i> Phase Visibility
```

Comment:
We need to bind the icon with the selected phase's phase visibility otherwise, it will show the ambiguous behaviour. Please see the recording below to see the ambiguity -

https://user-images.githubusercontent.com/44888949/123826460-695e5180-d91d-11eb-9fb4-08f27c14db76.mov